src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1386,15 +1390,20 @@ static void uct_ib_mlx5dv_check_dm_ksm_reg(uct_ib_mlx5_md_t *md)
         return;
     }
 
+    devx_memh = ucs_derived_of(memh, uct_ib_mlx5_devx_mem_t);
+    status    = uct_ib_mlx5_devx_reg_atomic_key(md, devx_memh);
+    if (status == UCS_OK) {
+        /* Enable device memory only if atomics are available*/
+        md->super.cap_flags |= UCT_MD_FLAG_ALLOC;
+        uct_ib_mlx5_devx_obj_destroy(devx_memh->atomic_dvmr, "DM-ATOMIC-KSM");
```

Comment:
shouldn't uct_ib_mlx5_devx_device_mem_free release atomic_dvmr?
if not, what happens in normal flow when user allocates DM, packs with atomic, and then releases the DM ?

Reply:
atomic_dvmr is not allocated within `uct_ib_mlx5_devx_device_mem_alloc`, so I thought it should be freed as part of `uct_ib_mlx5_devx_mem_dereg` (.mem_dereg) called from `ucp_mem_unmap`

Reply:
UCT level API user is calling `uct_md_mkey_pack_v2` if he wants to use atomics, and has to call `uct_md_mem_dereg_v2` accordingly to release the atomic / indirect keys created in `mkey_pack`.